src/backend/parser/gram.y

```diff
@@ -12467,6 +12481,81 @@ simple_select:
 				}
 		;
 
+hierarchical_query_clause:
+			{
```

Comment:
Should we keep the same as the existing code? i.e. keep the empty clause in the tail.